ocaml/lambda/matching.ml

```diff
@@ -1162,14 +1127,9 @@ let same_actions = function
 let safe_before ((p, ps), act_p) l =
   (* Test for swapping two clauses *)
   let same_actions act1 act2 =
-    match act1, act2 with
-    | Unguarded act1, Unguarded act2 ->
-        (match make_key act1, make_key act2 with
-         | Some key1, Some key2 -> key1 = key2
-         | None, _ | _, None -> false)
-    (* CR-soon rgodse: Investigate the callsites of this function to determine
-       if Guarded rhs's should ever be deemed the same. *)
-    | Guarded _, _ | _, Guarded _ -> false
+    match make_key (lambda_of_rhs act1), make_key (lambda_of_rhs act2) with
+    | Some key1, Some key2 -> key1 = key2
```

Comment:
I would think it's safest/clearest to classify the actions as the same only if they're both guarded RHSes with the same key *and* the same exn label.

Reply:
Done!